### PR TITLE
Separate transactionId and traceId

### PR DESCRIPTION
**Pull request by bram123**
Rename the 'request_id' to 'trace_id'
Add new `transaction_id`
- This ID is always generated for each new request/process and is not passed to other requests/processes.

This way we can group log messages by transaction (one http request) or by trace (a group of requests)

### CODE REVIEW

**Comment by bram123** [comment]
WIP + suggestion to rename the package to "symfony-trace-bundle"

**Comment by bartholdbos** [comment]
> WIP + suggestion to rename the package to "symfony-trace-bundle"

If we rename the package to symfony-trace-bundle it might be good to also follow best practises about bundle naming: https://symfony.com/doc/current/bundles/best_practices.html#bundle-name
Bundle class would be DigitalRevolutionSymfonyTraceBundle for example then

**Comment by frankdekker** [comment]
> > WIP + suggestion to rename the package to "symfony-trace-bundle"
> 
> If we rename the package to symfony-trace-bundle it might be good to also follow best practises about bundle naming: https://symfony.com/doc/current/bundles/best_practices.html#bundle-name Bundle class would be DigitalRevolutionSymfonyTraceBundle for example then

Well that's a mouth full, would recommend shorten DigitalRevolution to DR. 

Edit: Looking at the documentation, symfony recommends basing it on the namespace. It should then be `DRSymfonyTraceBundle`.

**Comment by bram123** [comment]
> WIP + suggestion to rename the package to "symfony-trace-bundle"

Renaming the package/bundle will be done in a separate PR.
My suggestion would be simply "TraceBundle", to keep in line with our other symfony bundles.
- And implement the getName() method to specify "dr_trace_bundle"

File: src/Monolog/TraceIdProcessor.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+declare(strict_types=1);
+
+namespace DR\SymfonyRequestId\Monolog;
+
+use DR\SymfonyRequestId\IdStorageInterface;
+use Monolog\LogRecord;
+use Monolog\Processor\ProcessorInterface;
+
+/**
+ * Adds the request ID to the Monolog record's `extra` key, so it can be used in formatters, etc.
```
Comment (frankdekker) [inline src/Monolog/TraceIdProcessor.php]: rename `request` to `trace`

File: src/DependencyInjection/SymfonyRequestIdExtension.php [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
-use DR\SymfonyRequestId\RequestIdStorageInterface;
+use DR\SymfonyRequestId\Monolog\TraceIdProcessor;
+use DR\SymfonyRequestId\IdGeneratorInterface;
+use DR\SymfonyRequestId\IdStorageInterface;
 use DR\SymfonyRequestId\SimpleIdStorage;
-use DR\SymfonyRequestId\Twig\RequestIdExtension;
+use DR\SymfonyRequestId\Twig\TraceIdExtension;
 use RuntimeException;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
 use Symfony\Component\DependencyInjection\Exception\LogicException;
```
Comment (frankdekker) [inline src/DependencyInjection/SymfonyRequestIdExtension.php:19]: Rename class

Comment (bram123) [reply src/DependencyInjection/SymfonyRequestIdExtension.php:19]: My idea was to do this in a separate PR, where the bundle is renamed and possibly also the git package

**Review by frankdekker — APPROVED** [review]

**Pull request merged**